### PR TITLE
IST-169 - Star Rating Block

### PR DESCRIPTION
**Pull request by vrundakansara**
### Description
Added a new block that can add Star Rating on page. Below are the options:
- Rating
- Range of rating
- Spacing options
- Color options
- Title beside the stars
- Typography and color options for title
- Other design options

### Types of changes
New Block

### How has this been tested?
- Test all the options mentioned in the settings.

### Checklist:
- [x] My code is tested
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [ ] I've included any necessary tests 
- [ ] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

**Review by Nikschavan — CHANGES_REQUESTED** [review]

File: src/blocks/star-rating/save.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+/**
+ * BLOCK: Star Rating - Save Block
+ */
+
+import classnames from "classnames"
+
+const {
+	RichText
+} = wp.blockEditor
```
[the remark this answers is not in the document]
Comment (vrundakansara) [reply src/blocks/star-rating/save.js:9]: This needs a global fix, everywhere we have used such packages. May be we can add a task for UAG here, and get it fixed all over in the plugin.

File: src/blocks/star-rating/edit.js [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+	RichText,
+	ColorPalette,
+} = wp.blockEditor
+
+const {
+	PanelBody,
+	SelectControl,
+	RangeControl,
+	BaseControl
+} = wp.components
```
[the remark this answers is not in the document]
Comment (vrundakansara) [reply src/blocks/star-rating/edit.js:30]: This needs a global fix, everywhere we have used such packages. Maybe we can add a task for UAG here, and get it fixed all over in the plugin.

File: src/blocks/star-rating/save.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (Nikschavan) [reply src/blocks/star-rating/save.js:9]: why not fix in the new code? old code can be refactored at a later priority.

File: src/blocks/star-rating/edit.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (Nikschavan) [reply src/blocks/star-rating/edit.js:30]: Same as above

File: src/blocks/star-rating/edit.js [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+					) }
+				>
+					<RichText
+						tagName="p"
+						placeholder={ __( "Write a title", 'ultimate-addons-for-gutenberg' ) }
+						value={ title }
+						className='uag-star-rating__title'
+						onChange={ ( value ) => setAttributes( { title: value } ) }
+					/>
+					<div class="uag-star-rating" title={`${rating}/${range}`} itemtype="http://schema.org/Rating" itemscope="" itemprop="reviewRating">
```
[the remark this answers is not in the document]
Comment (vrundakansara) [reply src/blocks/star-rating/edit.js]: As per our discussion, we are skipping this. Marking as resolved.

File: src/blocks/star-rating/edit.js [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+		super( ...arguments )
+	}
+
+	componentDidMount() {
+
+		// Assigning block_id in the attribute.
+		this.props.setAttributes( { block_id: this.props.clientId.substr( 0, 8 ) } )
+
+		// Pushing Style tag for this block css.
+		const $style = document.createElement( "style" )
```
[the remark this answers is not in the document]
Comment (vrundakansara) [reply src/blocks/star-rating/edit.js]: Yes, done. 👍🏼

File: src/blocks/star-rating/save.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (vrundakansara) [reply src/blocks/star-rating/save.js:9]: Done. 👍🏼

File: src/blocks/star-rating/edit.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (vrundakansara) [reply src/blocks/star-rating/edit.js]: Done. 👍🏼

**Review by Nikschavan — APPROVED** [review]

File: classes/class-uagb-config.php [hunk 5]
```diff
@@ -2921,6 +2921,37 @@ public static function get_block_attributes() {
 							'playOn'           => 'none',
 						),
 					),
+					'uagb/star-rating'            => array(
+						'slug'        => '',
+						'title'       => __( 'Star Rating', 'ultimate-addons-for-gutenberg' ),
+						'description' => __( 'This block helps you add Star Rating and customize it as required.', 'ultimate-addons-for-gutenberg' ),
+						'default'     => true,
+						'js_assets'   => array(),
+						'extension'   => false,
```
Comment (RajkiranBagal) [inline classes/class-uagb-config.php:2930]: I guess we can remove this extension key as we have removed it in [#764 ](https://github.com/brainstormforce/ultimate-addons-for-gutenberg/pull/764)PR.

File: src/blocks/star-rating/edit.js [hunk 6]
```diff
@@ -0,0 +1,264 @@
+/**
```
Comment (RajkiranBagal) [inline src/blocks/star-rating/edit.js:1]: Create a Functional Component, Create Separate Components Files for Settings & Rendering.

The lazy loading & linter changes we  will get those done afterwards.

**Review by RajkiranBagal — CHANGES_REQUESTED** [review]

File: classes/class-uagb-config.php (hunk 5 above)
Comment (vrundakansara) [reply classes/class-uagb-config.php:2930]: I think I need the key for this - https://github.com/brainstormforce/ultimate-addons-for-gutenberg/pull/764

Comment (vrundakansara) [reply classes/class-uagb-config.php:2930]: Done.

File: src/blocks/star-rating/edit.js (hunk 6 above)
Comment (vrundakansara) [reply src/blocks/star-rating/edit.js:1]: Done.

**Comment by vrundakansara** [comment]
Handled this PR here - https://github.com/brainstormforce/ultimate-addons-for-gutenberg/pull/807
Closing this.

**Pull request closed without merging**